zero/src/rpc/native/txn.rs

```diff
@@ -100,26 +197,16 @@ where
 }
 
 /// Fetches the transaction data for the given transaction hash.
-async fn fetch_tx_data<ProviderT, TransportT>(
+async fn fetch_tx_receipt<ProviderT, TransportT>(
     provider: &ProviderT,
     tx_hash: &B256,
-) -> anyhow::Result<(<Ethereum as Network>::ReceiptResponse, GethTrace, GethTrace), anyhow::Error>
+) -> anyhow::Result<<Ethereum as Network>::ReceiptResponse>
 where
     ProviderT: Provider<TransportT>,
     TransportT: Transport + Clone,
 {
-    let tx_receipt_fut = provider.get_transaction_receipt(*tx_hash);
-    let pre_trace_fut = provider.debug_trace_transaction(*tx_hash, prestate_tracing_options(false));
-    let diff_trace_fut = provider.debug_trace_transaction(*tx_hash, prestate_tracing_options(true));
-
-    let (tx_receipt, pre_trace, diff_trace) =
-        futures::try_join!(tx_receipt_fut, pre_trace_fut, diff_trace_fut,)?;
-
-    Ok((
-        tx_receipt.context("Transaction receipt not found.")?,
-        pre_trace,
-        diff_trace,
-    ))
+    let tx_receipt = provider.get_transaction_receipt(*tx_hash).await?;
+    Ok(tx_receipt.context("Transaction receipt not found.")?)
```

Comment:
```suggestion
    tx_receipt.ok_or(anyhow!("Transaction receipt not found."))
```

Reply:
`Ok(result?)` is definitely an antipattern.

I will go with `tx_receipt.context("Transaction receipt not found.")`.